src/common/form/toggle/index.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import classNames from 'classnames'
+
+import Component from '../../base-component'
+import Icon from '../../icon'
+import propTypes from '../../prop-types'
+
+import styles from './index.css'
+
+@propTypes({
+  className: propTypes.string,
+  defaultValue: propTypes.bool,
+  onChange: propTypes.func,
+  icon: propTypes.string,
+  iconOn: propTypes.string,
+  iconOff: propTypes.string,
+  iconSize: propTypes.number,
+  value: propTypes.bool
+})
+export default class Toggle extends Component {
+  static defaultProps = {
+    iconOn: 'toggle-on',
+    iconOff: 'toggle-off',
+    iconSize: 2
+  }
+
+  get value () {
+    const { props } = this
+
+    const { value } = props
+    if (value != null) {
+      return value
+    }
+
+    const { input } = this.refs
+    if (input) {
+      return input.checked
+    }
+
+    return props.defaultValue || false
+  }
+
+  set value (value) {
+    if (process.env.NODE_ENV !== 'production') {
+      if (this.props.value != null) {
```

Comment:
The original code was like that in purpose because I thought having a simple condition was necessary for the code minifier to prune the code depending on the environment but it looks like I was wrong therefore feel free merge the 2 conds.